integration/spark/app/src/test/java/io/openlineage/spark/agent/lifecycle/LibraryTest.java

```diff
@@ -173,12 +171,37 @@ void testRDDName(SparkSession spark) {
     assertThat(s).isEqualTo("map_partitions_numbers");
   }
 
+  @SneakyThrows
   private void verifySerialization(List<OpenLineage.RunEvent> events)
       throws JsonProcessingException {
     for (OpenLineage.RunEvent event : events) {
-      assertNotNull(
-          "Event can serialize",
-          OpenLineageClientUtils.newObjectMapper().writeValueAsString(event));
+      assertNotNull("Event can serialize", writeValueAsString(event));
     }
   }
+
+  // workaround to use OpenLineage client's ObjectMapper which is shaded
```

Comment:
I don't think you need any shading workarounds here - `OpenLineageClientUtils` is a direct dependency that will be shaded later, and you can access any of it's shaded dependencies with fully-qualified class names too.

Reply:
>So the shading makes OpenLineageClientUtils return a shaded ObjectMapper instead.

That does not sound like it is a problem - the import of ObjectMapper should be for the shaded version.

>It is even more complex, because the shading is applied after compiling whole source code. So, if we use here shaded version, the code will not compile as shaded version is not available at the compile stage. But it is required during test runtime.

OK, but this is shaded in the Java client part, which is separate from shading here? So I don't really understand why does it matter.

Reply:
Well, it took me some time to realise that you're right! Thanks. 
I got rid of reflection here.